lib/src/front_end/piece_factory.dart

```diff
@@ -828,17 +830,96 @@ mixin PieceFactory {
   }
 
   /// Creates a [ListPiece] for the given bracket-delimited set of elements.
-  Piece createList(Iterable<AstNode> elements,
+  ///
+  /// If [preserveNewlines] is `true`, then any newlines or lack of newlines
+  /// between pairs of elements in the input are preserved in the output. This
+  /// is used for collection literals that contain line comments to preserve
+  /// the author's deliberate structuring, as in:
+  ///
+  ///     matrix = [
+  ///       1, 2, 3,
```

Comment:
[nit] Show this "contain line comments" detail in the example.

```suggestion
  /// is used for collection literals that contain line comments to preserve
  /// the author's deliberate structuring, as in:
  ///
  ///     matrix = [
  ///       1, 2, 3, //
```